server/bot/commands/trivia.ts

```diff
@@ -0,0 +1,104 @@
+    argsRequired: true
+  },
+  generator: async (message: Message, args: string[], { tempDB, client }) => {
+    let botPlays = false
+    let timeLimit = 15000
+    let maxScore = 30
+    let revealAnswer = true
+
+    if (args.find(element => element.includes('--bot-plays='))) {
+      if (args.find(element => element.includes('--bot-plays=')).split('=')[1] === 'true') {
+        botPlays = true
+      } else if (args.find(element => element === '--bot-plays=').split('=')[1] !== 'false') {
+        return 'Invalid usage.'
+      }
+    }
+    if (args.find(element => element.includes('--time-limit='))) {
+      if (+args.find(element => element.includes('--time-limit=')).split('=')[1] > 4) {
+        timeLimit = +args.find(element => element.includes('--time-limit=')).split('=')[1]
+      } else {
+        return 'Invalid usage'
+      }
+    }
+    if (args.find(element => element.includes('--max-score='))) {
+      if (+args.find(element => element.includes('--max-score=')).split('=')[1] > 0) {
+        maxScore = +args.find(element => element.includes('--max-score')).split('=')[1]
+      } else {
+        return 'Invalid usage.'
+      }
+    }
+    if (args.find(element => element.includes('--reveal-answer='))) {
+      if (args.find(element => element.includes('--reveal-answer=')).split('=')[1] === 'false') {
+        revealAnswer = false
+      } else if (args.find(element => element === '--reveal-answer=').split('=')[1] !== 'true') {
+        return 'Invalid usage.'
+      }
+    }
+    if (args.length === 1 && ['scores', 'score', 'leaderboard'].includes(args[0])) {
+      const session = tempDB.trivia[message.channel.id]
+      if (session) {
+        message.channel.createMessage(await session.sendScores())
+      } else {
+        return 'There is no trivia session ongoing in this channel.'
+      }
+    } else if (args.length === 1 && args[0] === 'list') {
+      return 'Available trivia lists: \n\n-anime\n-artandliterature\n-cars\n-computers\n-disney\ndota2abilities\n-dota2items\n-elements\n-entertainment\n-finalfantasy\n-gameofthrones\n-games\n-general\n-greekmyth\n-harrypotter\n-leagueoflegends\n-leagueults\n-nba\n-overwatch\n-pokemon\n-slogans\n-sports\n-starwars\n-uscapitals\n-usflags\n-usmap\n-usstateabbreviations\n-warcraft\n-whosthatpokemon\n-worldcapitals\n-worldflgs\n-worldmap'
+    } else if (args.length === 1 && args[0] === 'stop') {
+      const session = tempDB.trivia[message.channel.id]
+      const channel = message.channel as GuildTextableChannel
+      if (session) {
+        if (message.author === session.author || channel.permissionsOf(message.author.id).has('manageMessages')) {
+          await session.endGame()
+          return 'Trivia stopped.'
+        } else {
+          return 'You are not authorized to do that.'
+        }
+      } else {
+        return 'There is no trivia session ongoing in this channel.'
+      }
+    } else if (args[0] !== ('scores' || 'score' || 'leaderboard' || 'stop' || 'list')) {
```

Comment:
What? This is the same as `args[0] !== 'list'` and this whole condition is unneeded since it's part of an else-if.

Reply:
I originally just wanted to make sure they were running the command to start a trivia session, not sure what I was thinking when I wrote that condition but anyway, it has been updated accordingly